Automattic/pocket-casts-ios
# Refactor WatchOS source interface

Fixes #23

Refactors the following:
- Root interface of the WatchOS app, InterfaceController,  to use SwiftUI.
- Entry point of the app from using Storyboard entry point to a SwiftUI App scene.
- Implements a placeholder notification to replace the one done in storyboards
- Refactors NavigationManager to control SwiftUI view states

I tried to reduce the number the changes to a minimum and keep the data objects structure ( a lot of singletons) exactly as they were to avoid deeper changes on the view models across the app.

On the future we may need to refactor the way ViewModels are created to avoid direct dependencies to singletons like `SourceManager`

## To test

- Start the App on the simulator using the Scheme `Pocket Casts WatchApp` and use a target which pairs an WatchOS device with an iOS device.
- Ensure that you are logged out when you start

### Test logged out state

| 1 | 2 | 
| - | - |
| ![Simulator Screenshot - Apple Watch Series 9 (45mm) - 2024-06-06 at 22 26 56](https://github.com/Automattic/pocket-casts-ios/assets/651601/d915ad27-c827-43bc-aaec-6e721428a5d8) | ![Simulator Screenshot - Apple Watch Series 9 (45mm) - 2024-06-07 at 13 13 05](https://github.com/Automattic/pocket-casts-ios/assets/651601/b10311ce-e9a2-4600-a203-3c7669a2e0fc) |

- Check if the root screen is show correctly and you can only access the Phone source option
- Tap on Phone and see if you can navigate
- Tap on Watch and see that you cannot navigate in
- Tap on Refresh Account
- Sign in on the main app ( device simulator associated with watch)
- Tap on Refresh Account
- See if the screen shows the correct user account

### Test login with a Plus account

| 1 | 2 | 
| - | - |
| ![Simulator Screenshot - Apple Watch Series 9 (45mm) - 2024-06-06 at 22 29 07](https://github.com/Automattic/pocket-casts-ios/assets/651601/eaae8a19-468b-403a-bf16-b9d25122afff)  | ![Simulator Screenshot - Apple Watch Series 9 (45mm) - 2024-06-06 at 22 29 11](https://github.com/Automattic/pocket-casts-ios/assets/651601/597f9977-543f-4681-ab7c-aaa8501c6ead) |

- Login on the main app out in the main app and using the Scheme `Pocket Casts WatchApp`
- Tap on Refresh Data
- Tap on Phone source
- Tap on

### Test Navigation

- On both sources, Phone and Watch, go inside the main options and smoke test them
  - Now Playing, 
  - UpNext, 
  - Podcasts, 
  - Filters, 
  - Downloads
  - Files 

### Test Restore

- On both sources, Phone and Watch, go inside the main options
- And then kill the app when inside one of the options
- Start the app again and see if the correct screen is restored
- This is only implemented for root level navigation items ( as before )

### Test Push Notififications

- Drag and Drop the `PushNotificationPayload.apns` file to the watch simulator
- Wait and see if the Push Notification shows correctly

## Checklist
